galaxy/models.py

```diff
@@ -312,67 +288,210 @@ def compute_users_clubs_score(cls, user1, user2) -> int:
     ###################
 
     @classmethod
-    def rule(cls) -> None:
-        GalaxyStar.objects.all().delete()
-        # The following is a no-op thanks to cascading, but in case that changes in the future, better keep it anyway.
-        GalaxyLane.objects.all().delete()
+    def scale_distance(cls, value) -> int:
+        # TODO: this will need adjustements with the real, typical data on Taiste
+        if value == 0:
+            return 4000  # Following calculus would give us +∞, we cap it to 4000
+
+        cls.logger.debug(f"\t\t> Score: {value}")
+        # Invert score to draw close users together
+        value = 1 / value  # Cannot be 0
+        value += 2  # We use log2 just below and need to stay above 1
+        value = (  # Let's get something in the range ]0; log2(3)-1≈0.58[ that we can multiply later
+            math.log2(value) - 1
+        )
+        value *= (  # Scale that value with a magic number to accommodate to typical data
+            # Really close galaxy citizen after 5 years typically have a score of about XXX
+            # Citizen that were in the same year without being really friends typically have a score of about XXX
+            # Citizen that have met once or twice only have a couple of pictures together typically score about XXX
+            cls.GALAXY_SCALE_FACTOR
+        )
+        cls.logger.debug(f"\t\t> Scaled distance: {value}")
+        return int(value)
+
+    def rule(self, picture_count_threshold=10) -> None:
+        total_time = time.time()
+        self.logger.info("Listing rulable citizen.")
         rulable_users = (
             User.objects.filter(subscriptions__isnull=False)
-            .filter(
-                Q(godchildren__isnull=False)
-                | Q(godfathers__isnull=False)
-                | Q(pictures__isnull=False)
-                | Q(memberships__isnull=False)
-            )
+            .annotate(pictures_count=Count("pictures"))
+            .filter(pictures_count__gt=picture_count_threshold)
+            .prefetch_related("memberships")
+            .prefetch_related("godfathers")
+            .prefetch_related("godchildren")
+            .prefetch_related("pictures")
```

Comment:
Why change filter with Q statements to a bunch of prefetch_related, even tough I don't see any use of those prefetched field thereafter ?

This seems to me like useless operations which add four extra queries to the database (whereas `select_related` uses a JOIN, `prefetch_relation` performs a separate lookup and joins the data in Python) while loosing the advantage of working on the smallest possible set of users.

Reply:
Indeed, that's a part of some performance tests I made a while ago and never got rid of it. Removed.